### ISSUE TITLE
Big preallocation slows tests

### ISSUE THREAD

**Issue by F3kilo** [dfinity/stable-structures#15]
In cargo tests, we want to provide independent memory, memory manager and data structures for each test. But for first allocation `MemoryManager` allocate a bucket with 1024 pages of memory (about 67Mb). That operation is slow and requires a lot of memory, which will never be used.
I think a library client need some control over first allocation size. For example, client may replace `const BUCKET_SIZE_IN_PAGES`, with custom value in `MemoryManager` constructor.

**Comment by ielashi** [comment]
Thanks for raising this @F3kilo. IIUC this problem can be solved by simply making [this method](https://github.com/dfinity/stable-structures/blob/main/src/memory_manager.rs#L142) public. Would that solve your use-case?

**Comment by F3kilo** [comment]
Thanks for quick answer, @ielashi. Yes, it would help.

**Comment by ielashi** [comment]
I looked into this more closely and added benchmarks to understand the impact of the bucket size of the MemoryManager on performance (see https://github.com/dfinity/stable-structures/pull/29). Based on the data, it seems that a smaller bucket size has a negligible impact on performance, so it'd make sense based on the data to reduce the current default bucket size from 1024 pages to 64 or 128 pages. A smaller default bucket size would be the cleaner way to resolve this issue, I believe.

Having said that, I don't think there's any reason not to give developers the ability to specify a custom bucket size, so #22 has now been merged (thanks @ufoscout!)
